src/shogun/mathematics/Statistics.cpp

```diff
@@ -814,6 +815,157 @@ SGMatrix<float64_t> CStatistics::sample_from_gaussian(SGVector<float64_t> mean,
 	return S;
 }
 
+CStatistics::SigmoidParamters CStatistics::fit_sigmoid(SGVector<float64_t> scores, SGVector<float64_t> labels,
+	index_t maxiter, float64_t minstep,
+	float64_t sigma, float64_t epsilon)
+{
+	SG_SDEBUG("entering CStatistics::fit_sigmoid()\n")
+ 
+	REQUIRE(scores.vector, "CStatistics::fit_sigmoid() requires "
+			"scores vector!\n");
+
+	/* count prior0 and prior1 if needed */
+	int32_t prior0=0;
+	int32_t prior1=0;
+	SG_SDEBUG("counting number of positive and negative labels\n")
+	{
+		prior1 = std::count_if(labels.begin(), labels.end(), [](float64_t label) {return label > 0;});
+		prior0 = labels.vlen - prior1;
+	}
+	SG_SDEBUG("%d pos; %d neg\n", prior1, prior0)
+
+	/* construct target support */
+	float64_t hiTarget=(prior1+1.0)/(prior1+2.0);
+	float64_t loTarget=1/(prior0+2.0);
+	index_t length=prior1+prior0;
+
+	SGVector<float64_t> t(length);
+	for (index_t i=0; i<length; ++i)
+	{
+		if (labels[i]>0)
+			t[i]=hiTarget;
+		else
+			t[i]=loTarget;
+	}
+
+	/* initial Point and Initial Fun Value */
+	/* result parameters of sigmoid */
+	float64_t a=0;
+	float64_t b=CMath::log((prior0+1.0)/(prior1+1.0));
+	float64_t fval=0.0;
+
+	for (index_t i=0; i<length; ++i)
+	{
+		float64_t fApB=scores[i]*a+b;
```

Comment:
if this code is copy pasted, fine. Otherwise, can we rename this variable? :)

Reply:
It is mostly copy-pasted. Should I rename it?

Reply:
maybe leave it so it is clear where the code is coming from